client/my-sites/email/add-mailboxes/index.tsx

```diff
@@ -358,14 +366,14 @@ const AddMailboxes = ( props: AddMailboxesProps ): JSX.Element | null => {
 
 	const goToEmail = (): void => {
 		let url = emailManagement(
-			selectedSite.slug,
+			selectedSite?.slug,
 			isSelectedDomainNameValid ? selectedDomainName : null,
 			currentRoute,
 			{ source }
 		);
 
 		if ( source === INBOX_SOURCE ) {
-			url = emailManagementInbox( selectedSite.slug );
+			url = emailManagementInbox( selectedSite?.slug );
```

Comment:
`emailManagementInbox()` already supports an undefined site slug so this just prevents a fatal error if there is no selected site.